### ISSUE TITLE
Custom oauth not created via docker-compose

### ISSUE THREAD

**Issue by mbassiouny33** [RocketChat/Rocket.Chat#15891]
### Steps to reproduce:

1. run the docker-compose file
2. go to rocketchat address

### Expected behavior:
custom oAuth should be created and activated.

### Actual behavior:
Nothing! the custom oauth is not created

### Server Setup Information:

- Version of Rocket.Chat Server: 2.0 since 2.2 has a bug with oauth
- Operating System: 
- Deployment Method: docker
- Number of Running Instances: 1
- NodeJS Version: default one of the docker image
- MongoDB Version:4

### Client Setup Information

- Desktop App or Browser Version:
- Operating System:

### Additional context

Rocket.chat 2.2 has an issue with oAuth. So I decided to use 2.0.  see : https://github.com/RocketChat/Rocket.Chat/issues/15876

Now with 2.2 adding a custom oAuth via the web interface works fine. But if I try to create a custom oAuth via environment variables it does not work.

Here's my docker-compose.yml file, based on the official doc and this [link](https://gitlab.medelexis.ch/elexis/elexis-environment/blob/b18cc1ca760f585a1eb88cdd008edc674b34e03f/docker/compose/rocketchat.yml): 

```
version: '2'

services:
  rocketchat:
    image: rocketchat/rocket.chat:2.0.0
    command: >
      bash -c
        "for i in `seq 1 30`; do
          node main.js &&
          s=$$? && break || s=$$?;
          echo \"Tried $$i times. Waiting 5 secs...\";
          sleep 5;
        done; (exit $$s)"
    restart: unless-stopped
    volumes:
      - ./uploads:/app/uploads
    environment:
      - PORT=3500
      - ROOT_URL=http://rk.domain.me:3500
      - MONGO_URL=mongodb://mongo:27017/rocketchat
      - MONGO_OPLOG_URL=mongodb://mongo:27017/local
      - MAIL_URL=smtp://smtp.email
      - ADMIN_USERNAME=admin
      - ADMIN_PASS=password
      - ADMIN_EMAIL=me@sdfsdf.com
      - OVERWRITE_SETTING_Show_Setup_Wizard=completed
      - Accounts_OAuth_Custom-Keycloak=true
      - Accounts_OAuth_Custom-Keycloak-url=https://keycloack.domain.me:9445/auth
      - Accounts_OAuth_Custom-Keycloak-token_path=/realms/master/protocol/openid-connect/token
      - Accounts_OAuth_Custom-Keycloak-token_sent_via=Header
      - Accounts_OAuth_Custom-Keycloak-identity_token_sent_via=header
      - Accounts_OAuth_Custom-Keycloak-identity_path=/realms/master/protocol/openid-connect/userinfo
      - Accounts_OAuth_Custom-Keycloak-authorize_path=/realms/master/protocol/openid-connect/auth
      - Accounts_OAuth_Custom-Keycloak-scope=openid
      - Accounts_OAuth_Custom-Keycloak-id=rocket_client
      - Accounts_OAuth_Custom-Keycloak-secret=5ff81b24-81fb-4f92-8a3d-c139ab4ecf64
      - Accounts_OAuth_Custom-Keycloak-login_style=redirect
      - Accounts_OAuth_Custom-Keycloak-button_label_text=Login via Keycloak
      - Accounts_OAuth_Custom-Keycloak-button_label_color="#FFFFFF"
      - Accounts_OAuth_Custom-Keycloak-button_color="#13679A"
      - Accounts_OAuth_Custom-Keycloak-username_field=preferred_username
      - Accounts_OAuth_Custom-Keycloak-merge_users=false
#       - OVERWRITE_SETTING_Show_Setup_Wizard=completed 
#       - HTTP_PROXY=http://proxy.domain.com
#       - HTTPS_PROXY=http://proxy.domain.com
    depends_on:
      - mongo
    ports:
      - 3500:3500
    labels:
      - "traefik.backend=rocketchat"
      - "traefik.frontend.rule=Host: tora.domain.me"

  mongo:
    image: mongo:4.0
    restart: unless-stopped
    volumes:
     - ./data/db:/data/db
     #- ./data/dump:/dump
    command: mongod --smallfiles --oplogSize 128 --replSet rs0 --storageEngine=mmapv1
    labels:
      - "traefik.enable=false"

  # this container's job is just run the command to initialize the replica set.
  # it will run the command and remove himself (it will not stay running)
  mongo-init-replica:
    image: mongo:4.0
    command: >
      bash -c
        "for i in `seq 1 30`; do
          mongo mongo/rocketchat --eval \"
            rs.initiate({
              _id: 'rs0',
              members: [ { _id: 0, host: 'localhost:27017' } ]})\" &&
          s=$$? && break || s=$$?;
          echo \"Tried $$i times. Waiting 5 secs...\";
          sleep 5;
        done; (exit $$s)"
    depends_on:
      - mongo

  # hubot, the popular chatbot (add the bot user first and change the password before starting this image)
  hubot:
    image: rocketchat/hubot-rocketchat:latest
    restart: unless-stopped
    environment:
      - ROCKETCHAT_URL=rocketchat:3500
      - ROCKETCHAT_ROOM=GENERAL
      - ROCKETCHAT_USER=bot
      - ROCKETCHAT_PASSWORD=botpassword
      - BOT_NAME=bot
  # you can add more scripts as you'd like here, they need to be installable by npm
      - EXTERNAL_SCRIPTS=hubot-help,hubot-seen,hubot-links,hubot-diagnostics
    depends_on:
      - rocketchat
    labels:
      - "traefik.enable=false"
    volumes:
      - ./scripts:/home/hubot/scripts
  # this is used to expose the hubot port for notifications on the host on port 3001, e.g. for hubot-jenkins-notifier
    ports:
      - 3001:8080

  #traefik:
  #  image: traefik:latest
  #  restart: unless-stopped
  #  command: >
  #    traefik
  #     --docker
  #     --acme=true
  #     --acme.domains='your.domain.tld'
  #     --acme.email='your@email.tld'
  #     --acme.entrypoint=https
  #     --acme.storagefile=acme.json
  #     --defaultentrypoints=http
  #     --defaultentrypoints=https
  #     --entryPoints='Name:http Address::80 Redirect.EntryPoint:https'
  #     --entryPoints='Name:https Address::443 TLS.Certificates:'
  #  ports:
  #    - 80:80
  #    - 443:443
  #  volumes:
  #    - /var/run/docker.sock:/var/run/docker.sock

```

**Comment by reetp** [comment]
@rocket-cat label add duplicate

**Comment by reetp** [comment]
@rocket-cat close

You have already said this is a duplicate of #15876

Closing in favour of that.

**Comment by mbassiouny33** [comment]
Nope, it's not the same thing! 
In #15876 I complained about the button not doing anything in version 2.2 via web interface. The issue is present whether you use docker or not.

My issue here, is using version 2.0 (which works perfectly with oAuth) but having issue getting the oAuth created via environment variables. 

That said, Could you please re-open it?

Thanks in advance

**Comment by reetp** [comment]
This is for Rocket.Chat 2.0.0 so even if it is a bug it will not get fixed. At best it will get a fix in 2.3.x but you need to test on that first.

Please see the guide for opening issues: https://rocket.chat/docs/contributing/reporting-issues

> Make sure you are running the latest version of Rocket.Chat. It’s very likely simply updating will resolve your issue.

If one of the devs feels it needs re-opening they will.

**Comment by mbassiouny33** [comment]
Thanks for your reply. Makes sense to leave it for Devs to re-open if needed.

Updating makes it even worse as the latest version has no functional custom oAuth feature at all.

I'm just trying to point out that in addition to custom oAuth not working in 2.2, the custom oAuth can not be created with env. vars in docker (since 2.0). 

That way, Devs can see that they have 2 distinct issues to fix in their later releases.

I will just leave my comment here for Devs to see it if they're ever planning to fix any of the rocket.chat issues.

**Comment by reetp** [comment]
> Nope, it's not the same thing!

Note you did say "I have the same thing" and cross referenced here.

https://github.com/RocketChat/Rocket.Chat/issues/15876#issuecomment-560097313 

Your choice of wording was a little confusing. 

Either way I suspect they are linked and they devs will be on the case.

**Comment by mbassiouny33** [comment]
Oops! sorry for the confusion. I actually had 2 issues, I cross-refernced here in my 2nd issue just to explain why I'm using an older version( since the newer one had a bug with UI)

Anyway the Admin UI issue is now solved in 2.3.2. I edited my post accordingly.
After further testing with version 2.3.2, I figured my issue is actually like this one
https://github.com/RocketChat/Rocket.Chat/issues/10420

The client is not created but when I create the client manually, it 's auto-filled with the settings I wrote in the docker-compose file.
